documentation/docs/developer_guide/coding_standards.md

```diff
@@ -1 +1,160 @@
+Whilst syntax/text formatting style is relatively trivial, it must still be adhered to. Some more significant general rules that should be followed are: 
+## The JAC project 
+The JAC project is a long term initiative to include CABLE as an inter-operable Land-Surface Model (LSM) within the Unified Model (UM). For several reasons, CABLE is coupled to the UM via its native LSM, JULES. From a coding standards perspective, both the UM and JULES developments are strictly goverened by the UK Met Office and thus CABLE's coding standards are somewhat constrained by JULES' coding standards. 
+
+The [JULES coding standards](http://jules-lsm.github.io/coding_standards/) are described in this comprehensive document.
+### 1. IO must be restricted to the top level
+In terms of the offline application of the model this means IO should be restricted to the offline/ directory. Coupled to the UM/JULES there should be no IO code in the CABLE directory. 
+
+### 2. Comment liberally
+Detrimental to any model is the lack of meaningful and/or outdated comments. 
+Liberal inline commenting is expected aswell as thorough commenting in the MarkDown. See [cable_roughness.F90](https://github.com/CABLE-LSM/CABLE/blob/main/src/science/roughness/cable_roughness.F90) for an example, which is then redered as ?????
+### 3. Avoid WHERE loops
+WHERE loops effectively replace DO/IF loops. In some circumstances these may improve readability. They certainly make it easier to write code. However they are nightmare to debug and hamper portability. Especially nested WHERE loops. 
+Moreover it will not pass JULES coding standards. 
+### 4. Variable names
+We are not so limited by the number of character spaces available as we once were. Developers are encouraged to name variables meaningfully. Of course commenting might make it clear what the variable corresponds to however it is helpful to be able to understand what is going on whilst reading a line of code.
+
+Please maintain globally consistent names of variables across files/subroutines/modules. 
+### 5. File names
+To include CABLE code identifiably across applications, please include the "_cbl" suffix in your filename **eg. egName_cbl.F90**. 
+
+Please maintain consistent naming between files/subroutines/modules. 
+### 6. MODULE names
+Everything should be contained within modules.
+Modules should be of the form "*_mod_cbl". Following from the above file egName_cbl.F90 **MODULE egName_mod_cbl**.
+
+### 7. SUBROUTINE names
+Subroutines should be of the form "*_cbl()". Following from the above file egName_cbl.F90 **SUBROUTINE egName_cbl()**.
+
+### 8. Miscellaneous
+ * The last character of a line MUST not be placed beyond column 80. 
+ * SAVE attributes will not be permitted.
+ * Intrinsic FORTRAN subroutines, functions, etc should be capitalized
+ * Subroutines start on the margin. As do headers. Indentation of two spaces
```

Comment:
Re-Declaration(s) ...... anything before the code actually begins I guess. The UM is littered with reference to headers. ! End header. etc. I guess I haven't really thought about C headers. I haven't touched it since year 1 in my PhD ~17 years ago.

Reply:
I've always thought this was called the preamble/non-executable statements/declaration section. Nevertheless, I think this could be avoided completely by being a bit more specific about what statements should start on the first line. Declarations, USE, TYPE, IMPLICIT, FUNCTION. What about CONTAINS?

Reply:
There seems to have been an era where a UKMO programmer put !End of header comments at the end of every "preamble". Whilst perhaps not strictly correct it is very useful to skip to that  !End of header line. Especially when some of the "headers" are ~2000 lines.

Change made